windows/toni_ronnko_dirent.h

```diff
@@ -339,7 +339,13 @@ static void dirent_set_errno(int error);
  */
 static _WDIR *_wopendir(const wchar_t *dirname) {
   _WDIR *dirp;
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
```

Comment:
Oy, this is such annoying code, we need to find an actual cross platform answer :)

Do you or @lemire happen to know where this comes from, so we can merge upstream?

We should consider adding the URL that source at the top of the file, as well, for future updates.

Reply:
(To be clear, this is good :) Sure as hell not blocking this PR on a speculative cross-platform library.)

Reply:
The "original" (which appears to be quite different from ours) is there https://github.com/tronkko/dirent

It seems that our version was reformatted and maybe changed in other ways.

I merged this version as what we had before was much older and possibly unmaintained. It would be better to sync with upstream, drop our custom version, and push the fix upstream.

I would do it, but I don't even know what WINAPI_FAMILY_PARTITION means.

Reply:
I'm a *little* concerned that there is already a `#if WINAPI_FAMILY_PARTIOTION(WINAPI_PARTITION_DESKTOP)` further down in the function, suggesting it has been tested in this configuration on at least one machine where `DWORD` was fine ... is it possible we'll break some people with this change?

Reply:
Presumably, the problem is because wcslen returns size_t and not DWORD. So there is an implicit cast from size_t (sometimes 64 bits) to DWORD (definitively 32 bits). We can either change the type here or do an explicit cast later.

In either case, you get a string length, so DWORD is safe... I mean if you directory file name does not fit in 4GB you have other problems.

Reply:
I have just created a PR in the upstream: https://github.com/tronkko/dirent/pull/33